packages/backend/src/core/balances/providers/ArbitrumBalanceProvider.ts

```diff
@@ -0,0 +1,199 @@
+import { getTokenByAssetId } from '@l2beat/config'
+import {
+  AssetId,
+  Bytes,
+  ChainId,
+  EthereumAddress,
+  UnixTime,
+} from '@l2beat/shared-pure'
+import { BigNumber, utils } from 'ethers'
+
+import { ArbitrumMulticallClient } from '../../../peripherals/arbitrum/multicall/ArbitrumMulticall'
+import { MulticallResponse } from '../../../peripherals/arbitrum/multicall/interfaces'
+import { BalanceRecord } from '../../../peripherals/database/BalanceRepository'
+import { EthereumClient } from '../../../peripherals/ethereum/EthereumClient'
+import { MulticallRequest } from '../../../peripherals/ethereum/MulticallClient'
+import { BalanceProvider, BalanceQuery } from '../BalanceProvider'
+
+const erc20Interface = new utils.Interface([
+  'function balanceOf(address account) view returns (uint256)',
+])
+
+const multicallInterface = new utils.Interface([
+  'function getEthBalance(address account) view returns (uint256)',
+])
+
+export class ArbitrumBalanceProvider implements BalanceProvider {
+  private readonly chainId = ChainId.ARBITRUM
+
+  constructor(
+    private readonly arbitrumClient: EthereumClient,
+    private readonly multiCallClient: ArbitrumMulticallClient,
+  ) {}
```

Comment:
I won't assert `chainId` from the chain client since it is a prop-drilling promise to extract actual network Id from provider. For now, it won't hurt us much, since we have only two chains with explicitly written utility classes we can't really mix up. Let's write some universal interface as a separate task regarding L2Assets similar to  `Etherescanlike` which will be functioning as the already present `EthereumClient` with universal flavor of `chainId`

Reply:
https://github.com/l2beat/l2beat/pull/1645/commits/927ee044dd586ecceea0862df50b2e080bb4c8fb
https://github.com/l2beat/l2beat/pull/1645/commits/542db3437df31a58ec83a5aa950d93d53e032978